src/authz-smart/smartHandler.ts

```diff
@@ -8,31 +8,130 @@ import {
     AuthorizationRequest,
     AuthorizationBundleRequest,
     AllowedResourceTypesForOperationRequest,
+    SystemOperation,
+    TypeOperation,
 } from 'fhir-works-on-aws-interface';
-import { SMARTConfig } from './smartConfig';
+import axios from 'axios';
+import { LaunchType, ScopeType, SMARTConfig } from './smartConfig';
 
 // eslint-disable-next-line import/prefer-default-export
 export class SMARTHandler implements Authorization {
+    static readonly SCOPE_REGEX = /(patient|user|system|launch)\/(\w+|\*)\.?(\w+|\*)?|(^launch$)/;
+
     private readonly version: number = 1.0;
 
     private readonly config: SMARTConfig;
 
     constructor(config: SMARTConfig) {
-        this.config = config;
-        if (this.config.version !== this.version) {
-            throw Error('Configuration version does not match handler version');
+        if (config.version !== this.version) {
+            throw Error('Authorization configuration version does not match handler version');
         }
+        this.config = config;
     }
 
     async isAuthorized(request: AuthorizationRequest): Promise<boolean> {
+        const decoded = decode(request.accessToken, { json: true }) || {};
+        const { aud, iss } = decoded;
+        // verify aud & iss
+        if (this.config.expectedAudValue !== aud || this.config.expectedIssValue !== iss) {
+            console.error('aud or iss is not matching');
+            return false;
+        }
+        // verify scope
+        let scopes: string[] = [];
+        if (this.config.scopeValueType === 'space' && typeof decoded[this.config.scopeKey] === 'string') {
+            scopes = decoded[this.config.scopeKey].split(' ');
+        } else if (this.config.scopeValueType === 'array' && Array.isArray(decoded[this.config.scopeKey])) {
+            scopes = decoded[this.config.scopeKey];
+        }
+        if (!this.isScopeSufficient(scopes, request.operation, request.resourceType)) {
+            console.error('scopes are insuffiecient');
+            console.error(`scopes: ${scopes}`);
+            console.error(`operation: ${request.operation}`);
+            console.error(`resourceType: ${request.resourceType}`);
+            return false;
+        }
+
+        // Verify token
+        if (this.config.authZUserInfoUrl) {
+            console.log('Posting to AuthZ for more customer information');
+            let result;
+            try {
+                result = await axios.post(
+                    this.config.authZUserInfoUrl,
+                    {},
+                    { headers: { Authorization: `Bearer ${request.accessToken}` } },
+                );
+            } catch (e) {
+                console.error('Post to authZUserInfoUrl failed');

```

Comment:
Can this be combined to one line?
`console.error('Post to authZUserInfoUrl failed', e);`